demos/machine_translation_demo/python/machine_translation_demo.py

```diff
@@ -204,18 +210,24 @@ def main(args):
     model = Translator(
         model_xml=args.model,
         model_bin=args.model.with_suffix(".bin"),
+        device=args.device,
         tokenizer_src=args.tokenizer_src,
         tokenizer_tgt=args.tokenizer_tgt,
         output_name=args.output_name
     )
-
-    if args.input:
-        def sentences():
-            for sentence in args.input:
+    input_data = args.input
+    if input_data and len(input_data) == 1 and input_data[0].endswith('.txt'):
```

Comment:
The `len(input_data) == 1` condition seems a bit arbitrary. IMHO, any elements of `input_data` that end with `.txt` should be treated as file names.

Reply:
You didn't fully fix this - you removed the condition, but you haven't added the support for treating the other elements of `input_data` (besides `input_data[0]`) as text file paths.

Reply:
It's not clear for me, which other elements needed to be supported. There are only two cases - when only path to .txt file is submitted and with the location in `input_data[0]`, or when input consists of several sentences. Iteration is the same for all options of '-input'.

Reply:
There could be multiple txt files, or txt files intermingled with sentences.

Reply:
Updated.